Modules/Calendar/Calendar/DatePicker/View/DatePickerViewController.swift

```diff
@@ -0,0 +1,133 @@
+import UIKit
+import Utils
+import Assets
+import Architecture
+import RxCocoa
+import RxSwift
+import RxDataSources
+import Models
+import OtherServices
+
+public typealias DatePickerStore = Store<DatePickerState, DatePickerAction>
+
+public class DatePickerViewController: UIViewController, Storyboarded {
+
+    public static var storyboardName = "Calendar"
+    public static var storyboardBundle = Assets.bundle
+
+    @IBOutlet var collectionView: UICollectionView!
+    @IBOutlet var headerStackView: UIStackView!
+    private var dataSource: DatePickerCollectionViewDataSource!
+    private var layout: UICollectionViewFlowLayout!
+
+    private var disposeBag = DisposeBag()
+    private var calendar = Calendar.autoupdatingCurrent
+    private var sizeClassChangedSubject = BehaviorSubject<Void>(value: ())
+
+    public var store: DatePickerStore!
+    var time: Time!
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupHeaderLabels()
+
+        view.backgroundColor = Color.backgroundPrimary.uiColor
+        headerStackView.backgroundColor = Color.backgroundPrimary.uiColor
+        collectionView.backgroundColor = Color.backgroundPrimary.uiColor
+        collectionView.showsHorizontalScrollIndicator = false
+        collectionView.decelerationRate = .fast
+
+        collectionView.register(DatePickerDayCollectionViewCell.nib,
+                                forCellWithReuseIdentifier: DatePickerDayCollectionViewCell.reuseIdentifier)
+
+        dataSource = DatePickerCollectionViewDataSource(collectionView: collectionView, currentlySelectedDate: time.now())
+        collectionView.delegate = dataSource
+        dataSource.delegate = self
+        layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .horizontal
+        layout.minimumLineSpacing = 0
+        collectionView.setCollectionViewLayout(layout, animated: false)
+
+        Driver.combineLatest(
+            store.select({ headerDays($0.preferences.value?.firstDayOfTheWeek ?? .monday) }).distinctUntilChanged(),
+            sizeClassChangedSubject.asDriver(onErrorJustReturn: ()),
+            resultSelector: { days, _ -> [DayOfTheWeek] in
+                return days
+            }
+        )
+            .mapTo(toHeaderTitles)
+            .subscribe(onNext: updateHeaderTitles)
+            .disposed(by: disposeBag)
```

Comment:
Something like this?
```suggestion
            store.select({ headerDays($0.preferences.value?.firstDayOfTheWeek ?? .monday) }).distinctUntilChanged(),
            sizeClassChangedSubject.asDriver(onErrorJustReturn: ()),
            resultSelector: { days, _ -> [DayOfTheWeek] in
                return days
            })
            .mapTo(toHeaderTitles)
            .subscribe(onNext: updateHeaderTitles)
            .disposed(by: disposeBag)
```